src/block/carousel/frontend-carousel.js

```diff
@@ -166,20 +191,56 @@ class _StackableCarousel {
 
 	nextSlide = () => {
 		let newSlide = this.currentSlide + 1
-		if ( newSlide > this.maxSlides() ) {
+
+		if ( this.infiniteScroll && newSlide > this.slideEls.length ) {
+			this.swapSlides( 'N' )
+			return
+		}
+
+		if ( ! this.infiniteScroll && newSlide > this.maxSlides() ) {
 			newSlide = this.slideOffset
 		}
 		this.goToSlide( newSlide )
 	}
 
 	prevSlide = () => {
 		let newSlide = this.currentSlide - 1
+
+		if ( this.infiniteScroll && newSlide < this.slideOffset ) {
+			this.swapSlides( 'P' )
+			return
+		}
+
 		if ( newSlide < this.slideOffset ) {
```

Comment:
Seeing the `if` condition in the `nextSlide`, would you also need `! this.infiniteScroll` in here?
```suggestion
		if ( ! this.infiniteScroll && newSlide < this.slideOffset ) {
```

Reply:
Technically, the `! this.infiniteScroll` check won’t be needed here because it is already handled by the first if statement in line 209. Both if statements (in line 209 and 214) have the condition`newSlide < this.slideOffset`, and the first if statement checks `this.infiniteScroll` so reaching the second if statement would imply that `this.infiniteScroll` is false.

In contrast to `nextSlide()`, the second condition for both if statements differs (`newSlide > this.slideEls.length` and `newSlide > this.maxSlides`) so it is necessary to add the `! this.infiniteScroll` to the second if statement.

But, I will add `! this.infiniteScroll` just to be sure.